crates/nu-cli/src/prompt_update.rs

```diff
@@ -107,12 +106,13 @@ pub(crate) fn update_prompt<'prompt>(
     // Now that we have the prompt string lets ansify it.
     // <133 A><prompt><133 B><command><133 C><command output>
     let left_prompt_string = if config.shell_integration {
-        match left_prompt_string {
-            Some(prompt_string) => Some(format!(
+        if let Some(prompt_string) = left_prompt_string {
+            Some(format!(
                 "{}{}{}",
                 PRE_PROMPT_MARKER, prompt_string, POST_PROMPT_MARKER
-            )),
-            None => left_prompt_string,
+            ))
+        } else {
+            left_prompt_string
```

Comment:
I think in this case `.map`/`.and_then` would actually be less confusing.

Reply:
Agreed and changed.